api/views/phones.py

```diff
@@ -640,6 +643,168 @@ def sms_status(request):
+    required=["to"],
+    properties={"to": openapi.Schema(type=openapi.TYPE_STRING)},
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@swagger_auto_schema(method="post", request_body=call_body)
+@decorators.api_view(["POST"])
+def call(request):
+    """
+    Make a call from the authenticated user's relay number.
+
+    """
+    if not flag_is_active(request, "outbound_phone"):
+        return response.Response({"outbound_phone": False}, status=404)
+    to = request.data.get("to", None)
+    if to is None:
+        raise exceptions.ValidationError("Missing 'to' parameter.")
+    real_phone = RealPhone.objects.get(user=request.user)
+    relay_number = RelayNumber.objects.get(user=request.user)
+    client = twilio_client()
+    client.calls.create(
+        twiml=f"<Response><Say>Dialing {to} ...</Say><Dial>{to}</Dial></Response>",
+        to=real_phone.number,
+        from_=relay_number.number,
+    )
+    return response.Response(status=200)
+
+
+message_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    properties={
+        "body": openapi.Schema(type=openapi.TYPE_STRING),
+        "destination": openapi.Schema(type=openapi.TYPE_STRING),
+    },
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@swagger_auto_schema(method="post", request_body=message_body)
+@decorators.api_view(["POST"])
+def post_message(request):
+    """
+    Send a message from the user's realy number.
+
+    POST params:
+        body: the body of the message
+        destination: E.164-formatted phone number
+
+    """
+    if not flag_is_active(request, "outbound_phone"):
+        return response.Response({"outbound_phone": False}, status=404)
+    relay_number = RelayNumber.objects.get(user=request.user)
+    body = request.data.get("body")
+    destination_number = request.data.get("destination")
+    client = twilio_client()
+    client.messages.create(from_=relay_number.number, body=body, to=destination_number)
+    return response.Response(
+        status=200,
+    )
+
+
+messages_body = openapi.Schema(
+    type=openapi.TYPE_OBJECT,
+    properties={"with": openapi.Schema(type=openapi.TYPE_STRING)},
+)
+
+
+@decorators.permission_classes([permissions.IsAuthenticated, HasPhoneService])
+@decorators.api_view(["GET"])
+def messages(request):
+    """
+    Get the user's messages.
+
+    Pass ?with=<E.164> parameter to filter the messages to only the ones sent between
+    the relay number and the <E.164> number.
+
+    Pass ?direction=inbound|outbound to filter the messages to only the inbound or
+    outbound messages.
+
+    """
+    if not flag_is_active(request, "outbound_phone"):
+        return response.Response({"outbound_phone": False}, status=404)
+    _with = request.data.get("with", None)
+    _direction = request.data.get("direction", None)
+    relay_number = RelayNumber.objects.get(user=request.user)
+
+    contact = None
+    if _with is not None:
```

Comment:
I'm skipping validation of `_with` because it is implied by looking it up as `InboundContact.inbound_number`, which is E.164 format.

I'm adding that if no match is found, a 400 error is returned.